group-generators/helpers/data-providers/utils/utils.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * @param context  Context
+ * @param request  Request function
+ * @param cursor  Request function
+ * @param retry Number of retry
+ */
+export const retryRequest = async (
+  context: any,
+  request: any,
+  cursor: any,
+  retry: number
+) => {
+  let error;
+  for (let i = 0; i < retry; i++) {
+    try {
+      return await request(context, cursor);
```

Comment:
Is this possible to have a retryRequest without the notion of "cursor"? Why is it coupled with "curosr"?

Reply:
Because my queries need a cursor (we need it to know from which number I'll start to fetch again the profiles. (maybe "cursor" is not the right name?))

For example if I got an error while getting the [30000;30050] profiles chunk i'll have to retryThe request at 30000 so the cursor will be 30000. If I don't provide a cursor it will try to fetch the [0;50] profiles chunk.

Reply:
You can send a "generic" function and you just retry this function. It will be more "layerized" and usable in the future. (here if you want an other argument you need to change this function)